step-capstone/src/components/TripList.js

```diff
@@ -1,60 +1,53 @@
 import React from 'react';
 import { Grid } from '@material-ui/core'
 import TripItemComponent from "./TripItemComponent"
-import Trip from "./Trip"
+import {FirebaseContext} from './Firebase';
 import {
   Switch,
   Route,
   Redirect
 } from "react-router-dom";
 
+
 class TripList extends React.Component{
+  static contextType = FirebaseContext;
 
   constructor(props){
     super(props);
-    this.state = {selectedTrip : null};
+    this.state = {
+      userId : sessionStorage.getItem("userId"),
+      selectedTrip : null,
+      trips : []};
     this.handleOpenTrip = this.handleOpenTrip.bind(this);
   }
 
-  getTrips(userId) {
-    //Test Data
-    //TODO: Replace test data with call to firestore method
-    const tripTestOne = {
-      id : 1,
-      title : "London trip", 
-      startDate : new Date (2020, 8, 10),
-      endDate : new Date (2020, 8, 15),
-      description : "lorem ipsum ", 
-    }
-    const tripTestTwo = {
-      id : 2,
-      title : "Spain trip", 
-      startDate : new Date (2020, 12, 10),
-      endDate : new Date (2020, 12, 15),
-      description : "lorem ipsum ", 
-    }
-    return [tripTestOne, tripTestTwo];
+  componentDidMount() {
+    let trips = [];
+    const reference = "/users/" + this.state.userId + "/trips";
+    this.context.getCollection(reference).then(tripList => {
+      tripList.forEach(trip => {
+        trips.push(trip)
+      })
+      this.setState({trips : trips})
+    })
+    .catch(error => {console.log("Error Getting Trips")});
   }
 
-  handleOpenTrip(tripID) {
-    this.setState({ selectedTrip: tripID })
+  handleOpenTrip(tripId) {
+    this.setState({selectedTrip : tripId})
+    sessionStorage.setItem("tripId",tripId);
   }
 
   render () {
-    const trips = this.getTrips();
+    const trips = this.state.trips;
     return (
       <div>
-        <Switch>
-          <Route path = "/trips/:tripId">
-            <Trip tripId = {this.state.selectedTrip}/>
-          </Route>
-        </Switch>
         {this.state.selectedTrip ? <Redirect to = {"/trips/"+this.state.selectedTrip}/> :
```

Comment:
My current thought is redirecting the trip can be done just using onClick function? Why do we need to use State for this?

Reply:
one reason to have it like this so in the future, if we wanted to be able to get refresh working, we could check if a trip was selected, even if nobody had actually clicked the button. basically, it makes it so the criteria for loading a trip is whether that trip has been selected, rather than if the user specifically clicked the button triggering onClick.

Reply:
I believe it had to do with the whole refresh issue zach was having. Basically when refreshing it would try to go to the trip URL without knowing which one was clicked on since it was essentially "restarting" so we need to store locally the trip the user is currently on to go back.